src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -172,6 +172,8 @@ export class EditorPart extends Part implements IEditorGroupsService, IEditorGro
 
 		this._partOptions = newPartOptions;
 
+		this.centeredLayoutWidget.setFixedWidth(this._partOptions.centeredLayoutFixedWidth ?? false);
```

Comment:
I think a little bit nicer would be to listen to the event from here:

https://github.com/microsoft/vscode/blob/4f6f7d50b9941ab4dcd6280f49584c1c66e7b17a/src/vs/workbench/browser/parts/editor/editorPart.ts#L859

And forward into the widget.